pkg/webhook/injection.go

```diff
@@ -71,6 +71,37 @@ func injectSidecarContainer(pod *corev1.Pod, config *Config, supportsNativeSidec
 	}
 }
 
+func (si *SidecarInjector) injectMetadataPrefetchSidecarContainer(pod *corev1.Pod, config *Config, supportsNativeSidecar bool) {
```

Comment:
either add unit tests directly to this method, or use the broader integration test that checks the whole flow.

Reply:
let's add unit tests directly, as the integration test that covers the entire webhook does not explicitly check the new pod spec.